{
  "ID": "cfe93318-a8a0-463e-a3d7-a6f82855014e",
  "Root": {
    "alive": true,
    "content": [
      "55f6066e-5a5a-4ece-b4f7-5a48662a9e03",
      "e5b7fa88-b9b2-4fa7-bce7-75cd811cc8b4",
      "8e412db0-e545-44b0-a2be-6519986cd504",
      "9e07917c-e7d0-428c-9e47-4d79b34f6097",
      "28cfef2f-b1f6-4ad3-b420-a7bbd551be58",
      "d6cdd365-434e-4edb-996a-d58c61887a6f",
      "cec5bf3a-5e6a-472b-8b6e-f82e6d4c9bc4",
      "c76c4e56-a5b7-4b12-947e-d47b89b43c1b",
      "53523490-7f13-4285-a883-59baac18e2d5",
      "2f0a1314-e935-42e4-a109-d2832998e98e",
      "176b056f-a33f-4fa3-bbdc-09f9651b5d0f",
      "7e5aa81f-b4c9-4a40-a44b-1c1a5dd43ff2",
      "0787de6c-277f-4f58-ac6a-f58f43882694"
    ],
    "created_by": "bb760e2d-d679-4b64-b2a9-03005b21870a",
    "created_time": 1550393640000,
    "format": {
      "page_full_width": true,
      "page_small_text": true
    },
    "id": "cfe93318-a8a0-463e-a3d7-a6f82855014e",
    "last_edited_by": "bb760e2d-d679-4b64-b2a9-03005b21870a",
    "last_edited_time": 1550478060000,
    "parent_id": "12e6f78e-68a5-4972-90c9-6e1365ae6259",
    "parent_table": "block",
    "permissions": [
      {
        "role": "editor",
        "type": "user_permission",
        "user_id": "bb760e2d-d679-4b64-b2a9-03005b21870a"
      }
    ],
    "properties": {
      "title": [
        [
          "Raise Custom Errors Exceptions"
        ]
      ]
    },
    "type": "page",
    "version": 69,
    "content_resolved": [
      {
        "alive": true,
        "created_by": "bb760e2d-d679-4b64-b2a9-03005b21870a",
        "created_time": 1550393669022,
        "id": "55f6066e-5a5a-4ece-b4f7-5a48662a9e03",
        "ignore_block_count": true,
        "last_edited_by": "bb760e2d-d679-4b64-b2a9-03005b21870a",
        "last_edited_time": 1550478000000,
        "parent_id": "cfe93318-a8a0-463e-a3d7-a6f82855014e",
        "parent_table": "block",
        "properties": {
          "title": [
            [
              "Python has many built-in exceptions which force your program to output an error when something in it goes wrong."
            ]
          ]
        },
        "type": "text",
        "version": 3,
        "inline_content": [
          {
            "Text": "Python has many built-in exceptions which force your program to output an error when something in it goes wrong."
          }
        ]
      },
      {
        "alive": true,
        "created_by": "bb760e2d-d679-4b64-b2a9-03005b21870a",
        "created_time": 1550393669022,
        "id": "e5b7fa88-b9b2-4fa7-bce7-75cd811cc8b4",
        "ignore_block_count": true,
        "last_edited_by": "bb760e2d-d679-4b64-b2a9-03005b21870a",
        "last_edited_time": 1550478000000,
        "parent_id": "cfe93318-a8a0-463e-a3d7-a6f82855014e",
        "parent_table": "block",
        "properties": {
          "title": [
            [
              "However, sometimes you may need to create custom exceptions that serve your purpose."
            ]
          ]
        },
        "type": "text",
        "version": 3,
        "inline_content": [
          {
            "Text": "However, sometimes you may need to create custom exceptions that serve your purpose."
          }
        ]
      },
      {
        "alive": true,
        "created_by": "bb760e2d-d679-4b64-b2a9-03005b21870a",
        "created_time": 1550393669022,
        "id": "8e412db0-e545-44b0-a2be-6519986cd504",
        "ignore_block_count": true,
        "last_edited_by": "bb760e2d-d679-4b64-b2a9-03005b21870a",
        "last_edited_time": 1550478060000,
        "parent_id": "cfe93318-a8a0-463e-a3d7-a6f82855014e",
        "parent_table": "block",
        "properties": {
          "title": [
            [
              "In Python, users can define such exceptions by creating a new class. This exception class has to be derived, either directly or indirectly, from Exception class. Most of the built-in exceptions are also derived from this class."
            ]
          ]
        },
        "type": "text",
        "version": 5,
        "inline_content": [
          {
            "Text": "In Python, users can define such exceptions by creating a new class. This exception class has to be derived, either directly or indirectly, from Exception class. Most of the built-in exceptions are also derived from this class."
          }
        ]
      },
      {
        "alive": true,
        "created_by": "bb760e2d-d679-4b64-b2a9-03005b21870a",
        "created_time": 1550393669840,
        "id": "9e07917c-e7d0-428c-9e47-4d79b34f6097",
        "ignore_block_count": true,
        "last_edited_by": "bb760e2d-d679-4b64-b2a9-03005b21870a",
        "last_edited_time": 1550478000000,
        "parent_id": "cfe93318-a8a0-463e-a3d7-a6f82855014e",
        "parent_table": "block",
        "properties": {
          "title": [
            [
              "Custom Exception"
            ]
          ]
        },
        "type": "header",
        "version": 15,
        "inline_content": [
          {
            "Text": "Custom Exception"
          }
        ]
      },
      {
        "alive": true,
        "created_by": "bb760e2d-d679-4b64-b2a9-03005b21870a",
        "created_time": 1550393669837,
        "id": "28cfef2f-b1f6-4ad3-b420-a7bbd551be58",
        "ignore_block_count": true,
        "last_edited_by": "bb760e2d-d679-4b64-b2a9-03005b21870a",
        "last_edited_time": 1550478000000,
        "parent_id": "cfe93318-a8a0-463e-a3d7-a6f82855014e",
        "parent_table": "block",
        "properties": {
          "title": [
            [
              "Here, we have created a user-defined exception called CustomError which is derived from the Exception class. This new exception can be raised, like other exceptions, using the raise statement with an optional error message."
            ]
          ]
        },
        "type": "text",
        "version": 4,
        "inline_content": [
          {
            "Text": "Here, we have created a user-defined exception called CustomError which is derived from the Exception class. This new exception can be raised, like other exceptions, using the raise statement with an optional error message."
          }
        ]
      },
      {
        "alive": true,
        "created_by": "bb760e2d-d679-4b64-b2a9-03005b21870a",
        "created_time": 1550393669840,
        "id": "d6cdd365-434e-4edb-996a-d58c61887a6f",
        "ignore_block_count": true,
        "last_edited_by": "bb760e2d-d679-4b64-b2a9-03005b21870a",
        "last_edited_time": 1550478000000,
        "parent_id": "cfe93318-a8a0-463e-a3d7-a6f82855014e",
        "parent_table": "block",
        "properties": {
          "language": [
            [
              "Python"
            ]
          ],
          "title": [
            [
              "class CustomError(Exception):\n       pass\n\nx = 1\n\nif x == 1:\n    raise CustomError('This is custom error')"
            ]
          ]
        },
        "type": "code",
        "version": 5,
        "code": "class CustomError(Exception):\n       pass\n\nx = 1\n\nif x == 1:\n    raise CustomError('This is custom error')",
        "code_language": "Python"
      },
      {
        "alive": true,
        "created_by": "bb760e2d-d679-4b64-b2a9-03005b21870a",
        "created_time": 1550393669840,
        "id": "cec5bf3a-5e6a-472b-8b6e-f82e6d4c9bc4",
        "ignore_block_count": true,
        "last_edited_by": "bb760e2d-d679-4b64-b2a9-03005b21870a",
        "last_edited_time": 1550478000000,
        "parent_id": "cfe93318-a8a0-463e-a3d7-a6f82855014e",
        "parent_table": "block",
        "properties": {
          "title": [
            [
              "Output:"
            ]
          ]
        },
        "type": "text",
        "version": 4,
        "inline_content": [
          {
            "Text": "Output:"
          }
        ]
      },
      {
        "alive": true,
        "created_by": "bb760e2d-d679-4b64-b2a9-03005b21870a",
        "created_time": 1550393669840,
        "id": "c76c4e56-a5b7-4b12-947e-d47b89b43c1b",
        "ignore_block_count": true,
        "last_edited_by": "bb760e2d-d679-4b64-b2a9-03005b21870a",
        "last_edited_time": 1550478000000,
        "parent_id": "cfe93318-a8a0-463e-a3d7-a6f82855014e",
        "parent_table": "block",
        "properties": {
          "language": [
            [
              "Plain Text"
            ]
          ],
          "title": [
            [
              "Traceback (most recent call last):\n  File \"error_custom.py\", line 8, in \u003cmodule\u003e\n    raise CustomError('This is custom error')\n__main__.CustomError: This is custom error"
            ]
          ]
        },
        "type": "code",
        "version": 4,
        "code": "Traceback (most recent call last):\n  File \"error_custom.py\", line 8, in \u003cmodule\u003e\n    raise CustomError('This is custom error')\n__main__.CustomError: This is custom error",
        "code_language": "Plain Text"
      },
      {
        "alive": true,
        "created_by": "bb760e2d-d679-4b64-b2a9-03005b21870a",
        "created_time": 1550393671168,
        "id": "53523490-7f13-4285-a883-59baac18e2d5",
        "ignore_block_count": true,
        "last_edited_by": "bb760e2d-d679-4b64-b2a9-03005b21870a",
        "last_edited_time": 1550478000000,
        "parent_id": "cfe93318-a8a0-463e-a3d7-a6f82855014e",
        "parent_table": "block",
        "properties": {
          "title": [
            [
              "Catch custom Exception"
            ]
          ]
        },
        "type": "header",
        "version": 15,
        "inline_content": [
          {
            "Text": "Catch custom Exception"
          }
        ]
      },
      {
        "alive": true,
        "created_by": "bb760e2d-d679-4b64-b2a9-03005b21870a",
        "created_time": 1550393671165,
        "id": "2f0a1314-e935-42e4-a109-d2832998e98e",
        "ignore_block_count": true,
        "last_edited_by": "bb760e2d-d679-4b64-b2a9-03005b21870a",
        "last_edited_time": 1550478000000,
        "parent_id": "cfe93318-a8a0-463e-a3d7-a6f82855014e",
        "parent_table": "block",
        "properties": {
          "title": [
            [
              "This example shows how to catch custom Exception"
            ]
          ]
        },
        "type": "text",
        "version": 4,
        "inline_content": [
          {
            "Text": "This example shows how to catch custom Exception"
          }
        ]
      },
      {
        "alive": true,
        "created_by": "bb760e2d-d679-4b64-b2a9-03005b21870a",
        "created_time": 1550393671167,
        "id": "176b056f-a33f-4fa3-bbdc-09f9651b5d0f",
        "ignore_block_count": true,
        "last_edited_by": "bb760e2d-d679-4b64-b2a9-03005b21870a",
        "last_edited_time": 1550478000000,
        "parent_id": "cfe93318-a8a0-463e-a3d7-a6f82855014e",
        "parent_table": "block",
        "properties": {
          "language": [
            [
              "Python"
            ]
          ],
          "title": [
            [
              "class CustomError(Exception):\npass\n\ntry:\nraise CustomError('Can you catch me ?')\nexcept CustomError as e:\nprint ('Catched CustomError :{}'.format(e))\nexcept Exception as e:\nprint ('Generic exception: {}'.format(e))"
            ]
          ]
        },
        "type": "code",
        "version": 5,
        "code": "class CustomError(Exception):\npass\n\ntry:\nraise CustomError('Can you catch me ?')\nexcept CustomError as e:\nprint ('Catched CustomError :{}'.format(e))\nexcept Exception as e:\nprint ('Generic exception: {}'.format(e))",
        "code_language": "Python"
      },
      {
        "alive": true,
        "created_by": "bb760e2d-d679-4b64-b2a9-03005b21870a",
        "created_time": 1550393671168,
        "id": "7e5aa81f-b4c9-4a40-a44b-1c1a5dd43ff2",
        "ignore_block_count": true,
        "last_edited_by": "bb760e2d-d679-4b64-b2a9-03005b21870a",
        "last_edited_time": 1550478000000,
        "parent_id": "cfe93318-a8a0-463e-a3d7-a6f82855014e",
        "parent_table": "block",
        "properties": {
          "title": [
            [
              "Output:"
            ]
          ]
        },
        "type": "text",
        "version": 4,
        "inline_content": [
          {
            "Text": "Output:"
          }
        ]
      },
      {
        "alive": true,
        "created_by": "bb760e2d-d679-4b64-b2a9-03005b21870a",
        "created_time": 1550393671168,
        "id": "0787de6c-277f-4f58-ac6a-f58f43882694",
        "ignore_block_count": true,
        "last_edited_by": "bb760e2d-d679-4b64-b2a9-03005b21870a",
        "last_edited_time": 1550478000000,
        "parent_id": "cfe93318-a8a0-463e-a3d7-a6f82855014e",
        "parent_table": "block",
        "properties": {
          "language": [
            [
              "Plain Text"
            ]
          ],
          "title": [
            [
              "Catched CustomError :Can you catch me ?"
            ]
          ]
        },
        "type": "code",
        "version": 4,
        "code": "Catched CustomError :Can you catch me ?",
        "code_language": "Plain Text"
      }
    ],
    "title": "Raise Custom Errors Exceptions",
    "format_page": {
      "page_cover": "",
      "page_cover_position": 0,
      "page_font": "",
      "page_full_width": true,
      "page_icon": "",
      "page_small_text": true
    }
  },
  "Users": [
    {
      "email": "kkowalczyk@gmail.com",
      "family_name": "Kowalczyk",
      "given_name": "Krzysztof",
      "id": "bb760e2d-d679-4b64-b2a9-03005b21870a",
      "locale": "en",
      "mobile_onboarding_completed": true,
      "onboarding_completed": true,
      "profile_photo": "https://s3-us-west-2.amazonaws.com/public.notion-static.com/2dcaa66c-7674-4ff6-9924-601785b63561/head-bw-640x960.png",
      "time_zone": "America/Los_Angeles",
      "version": 18
    }
  ],
  "Tables": null
}